common/variant/src/main/java/org/apache/spark/types/variant/Variant.java

```diff
@@ -89,6 +97,37 @@ public BigDecimal getDecimal() {
     return VariantUtil.getDecimal(value, pos);
   }
 
+  // Get a raw date value (number of days from the Unix epoch) from the variant.
+  public int getRawDate() {
```

Comment:
do we need to have these methods? date/timestamp are logical types and are physically int/long. We just need the metadata to indicate them.

Reply:
Similarly, `UnsafeRow` in Spark doesn't have these methods either. The call just call `getInt/getLong` and interpret the result as number of days/seconds from Unix epoch.

Reply:
Thanks for the suggestion. This simplifies the implementation a lot.